packages/shared/src/components/auth/PostAuthBanner.tsx

```diff
@@ -1,11 +1,110 @@
 import React, { ReactElement } from 'react';
+import { useSearchParams } from 'next/navigation';
+import { useQuery } from '@tanstack/react-query';
 import { feature } from '../../lib/featureManagement';
 import { checkIsBrowser, checkIsExtension, UserAgent } from '../../lib/func';
 import { AuthExtensionBanner } from './AuthExtensionBanner';
 import { AuthenticationBanner } from './AuthenticationBanner';
+import { getBasicUserInfo } from '../../graphql/users';
+import { OnboardingHeadline } from './OnboardingHeadline';
+import { sizeClasses } from '../ProfilePicture';
+import { Image, ImageType } from '../image/Image';
 import { useConditionalFeature } from '../../hooks';
+import { IconSize } from '../Icon';
+import {
+  getSocialReferrer,
+  socialCTA,
+  socialGradient,
+  socialIcon,
+} from '../../lib/socialMedia';
+import { capitalize } from '../../lib/strings';
+import { useAuthContext } from '../../contexts/AuthContext';
+
+const geoToEmoji = (geo: string): string => {
+  return geo
+    .toUpperCase()
+    .split('')
+    .map((char) => String.fromCodePoint(char.charCodeAt(0) + 0x1f1a5))
+    .join('');
+};
+
+const GeoPersonalizedBanner = ({ geo }: { geo: string }): ReactElement => {
+  const emoji = geoToEmoji(geo);
+  const displayNames = new Intl.DisplayNames(['en'], { type: 'region' });
+  const country = displayNames.of(geo.toUpperCase());
+
+  return (
+    <AuthenticationBanner>
+      <span className="text-[3.5rem] leading-none">{emoji}</span>
+      <OnboardingHeadline
+        className={{
+          title: `typo-mega3`,
+          description: 'mb-8 typo-title3',
+        }}
+        title={`daily.dev is the fastest growing developer platform in ${country}!`}
+        description="We know how hard it is to be a developer. It doesn't have to be. Personalized news feed, dev community and search, much better than what's out there. Maybe ;)"
+      />
+    </AuthenticationBanner>
+  );
+};
```

Comment:
I think we can move along these components to their own separate file. Maybe just create a folder for example `banner` or `postBanner` to contain everything in there.

Reply:
@sshanzel I thought about it, but also didn't mind leaving them there because the PostAuthBanner doesn't even total 150 lines with all of them there. Don't see any other re-use for them currently.

Reply:
Yeah, this was more of a non-blocking. Was just suggestion.